### PR TITLE
Circular queue java

### PR DESCRIPTION
**Pull request by khusheekapoor**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have read the Contributing Guidelines on Pull Requests

### Description

I have written a code to implement basic operations (inserting and deleting an element, displaying) on Circular Queues in Java as a GSSoC'21 participant.

### Checklist

- [Yes] I've read the contribution guidelines.
- [Yes] I've checked the issue list before deciding what to submit.
- [yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

For issue #5042

### CODE REVIEW

File: Java/ds/CircularQueue.java [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+  * Time Complexity:
+  * insert - O(1)
+  * delete - O(1)
+  * display - O(n)
+  * 
+  * Space Complexity:
+  * O(n)
+  * 
+  * where n is number of elements
+ */
```
Comment (HarshCasper) [inline Java/ds/CircularQueue.java]: Add a new line at the EOF

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Can you give a smaller I/O sample?

**Comment by khusheekapoor** [comment]
> Can you give a smaller I/O sample?

Sure! Should I change the code (as in remove the menu), or only change the output in the comments (and mention that this is the essence of it)? @HarshCasper

**Comment by HarshCasper** [comment]
> > Can you give a smaller I/O sample?
> 
> Sure! Should I change the code (as in remove the menu), or only change the output in the comments (and mention that this is the essence of it)? @HarshCasper

Only the output

File: Java/ds/CircularQueue.java (hunk 1 above)
Comment (khusheekapoor) [reply Java/ds/CircularQueue.java]: Made the changes.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**